### ISSUE TITLE
Perfomance issue onLoad on "larger" page

### ISSUE THREAD

**Issue by ppumkin** [rmm5t/jquery-timeago#248]
I have a page load with a table, in the morning its about 40 elements, towards the end of the day it grows to 200.

I have other scripts running there but I tracked down that timeago **"blocks"** the page until all the elements are updated...  :-1: 

**"blocks"** - Cannot scroll the page or click on any element for 1-3 seconds

I think some optimization may be required as the main page does not mention any limits.
This page could potentailly shows thousands of rows with timago applie

**Comment by rmm5t** [comment]
It sounds like this is a dynamically updating page. Is it possible that this page is repeatedly/recursively calling timeago on the same elements over and over again?

**Comment by ppumkin** [comment]
No. I load the table Dom then on load the plugin. Afterwards not again,
ever.

**Comment by rmm5t** [comment]
>  in the morning its about 40 elements, towards the end of the day it grows to 200.

> No. I load the table Dom then on load the plugin.

This is a bit cryptic. If you're manually loading a table DOM, I don't understand how _"no"_ is the answer to whether you're dynamically updating the page DOM?

**Comment by ppumkin** [comment]
During the day the result set grows. The html is built in each refresh. No
Ajax is used. So early there's few rows later up to 200rows in table. It's
manual refresh only.
